archfx_cloud/api/connection.py

```diff
@@ -311,53 +217,70 @@ def __init__(self, domain=None, token_type=None, verify=True, timeout=None, retr
 
         if retries is not None or timeout is not None:
             adapter = _TimeoutHTTPAdapter(max_retries=retries, timeout=timeout)
-            self.session.mount('https://', adapter)
-            self.session.mount('http://', adapter)
+            self.session.mount("https://", adapter)
+            self.session.mount("http://", adapter)
+
+    def _destroy_tokens(self):
+        self.token = None
+        self.refresh_token_data = None
+        self.session.headers.pop("Authorization", "")
+
+    def _validate_and_set_tokens(self, data):
+        success = False
+        if isinstance(data, str):
+            self.token = data
+            success = True
+        elif "token" in data:
+            self.token = data["token"]
+            success = True
+        elif "access" in data:
```

Comment:
You are making an assumption here that we will want to return the data as 
```
"jwt": {
      "access"...
      "refresh"....
}
```

which does not necessary need to be true. The fact is that if we later worry that we want to be backwards compatible, we will probably instead return

```
"jwt": "access-key",
"refresh-jwt": "refresh-key"
```

I like your proposal better, but it does break any user out there (not using our package). We do have at least two customers using our API today (not likely using "refresh" but for sure using the "jwt" token

Reply:
The difference goes farther than that. SimpleJWT requires using separate refresh token to refresh access token (this is more secure - probably JWT spec was changed to require this since the old JWT plugin was released). So attempt to refresh the token using SimpleJWT-generated access token will fail.

Another thing is that `api-jwt-refresh/` view will return the same `{"access": "x", "refresh: "y"}` payload - it's easier to just use the same format.

Reply:
..but since not many people use the refresh (if any), and I would guess most people use `auth/login` instead of `auth-jwt-login` - we can have best of both worlds. "jwt" will still be a string, but presense of "jwt-refresh" would signal SimpleJWT and different handling of the refresh it requires.

Reply:
> The difference goes farther than that. SimpleJWT requires using separate refresh token to refresh access token

I understand. I am simply debating the format of your payload returning the two tokens. You are using

```
"jwt": {
      "access": "access-token"
      "refresh": "refresh-token"
}
```

while I am saying that for backwards compatibility (yes, just for the normal login part), we may need to use

```
"jwt": "access-token",
"refresh-jwt": "refresh-token"
```

in both cases, we are returning two different tokens for the user to keep. One (the "access-token") that everybody needs, and the second ("refresh-token") that only people that need to call refresh need (only web app)

And no, most people will use `auth/login` as that is all that we document. `auth-jwt-login` is basically advanced/hidden 

> we can have best of both worlds. "jwt" will still be a string, but presense of "jwt-refresh" would signal SimpleJWT and different handling of the refresh it requires.

Yes, that is exactly what I was saying. It is just that I actually do like using the `dict` in the long run, as it is clean. So just was saying that I don't yet know if I can just change it and have the customer update there scripts (the good think is that I know their names) in the name of the greater long term good, or if I want to make it a little uglier but stay backwards compatible.

My only point is that you are going backwards here. It is not about the python package, but about what we want the feature to look like.

I know you want to just clean up the broker now, but there is no real reason why we need to do that right now. we can wait a few weeks until we decide on what the cloud should look like, and then just change the broker and python package to match that. 

Again, I am willing to be convinced that we should just approve this for now  so the broker can be cleaned up now, but either way, we will still need to do more work in the future:

1. Update the Broker to match whatever the final cloud implementation is
2. Clean up the python package once we no longer need to support two packages

Reply:
I made the login backwards-compatible. It was pretty easy to do in retrospect :)

Reply:
https://github.com/iotile/archfx_broker/pull/1195 is the SimpleJWT swap - it's all in `server/`.

Tests are not actually changed that much - they just broken out into testing as close to a single thing as possible.